storage/backup/backup-cli/src/coordinators/restore.rs

```diff
@@ -133,8 +133,11 @@ impl RestoreCoordinator {
                 );
                 metadata_view.expect_state_snapshot(version)?
             } else {
+                let max_txn_ver = metadata_view
+                    .max_transaction_version()?
+                    .ok_or_else(|| anyhow!("No transaction backup found."))?;
                 metadata_view
-                    .select_state_snapshot(self.target_version())?
+                    .select_state_snapshot(std::cmp::min(self.target_version(), max_txn_ver))?
```

Comment:
Is that guaranteed a snapshot version at `max_txn_ver`?

Reply:
no, select_state_snapshot() finds one before or at it